### ISSUE TITLE
Try to use rygel instead of gmediarender for UPNP

### ISSUE THREAD

**Issue by mihaipora** [woutervanwijk/Pi-MusicBox#107]
TL;DR; : I love the current beta, but I think that there is a way to make it better by rendering in MPD using rygel.

My use-case is that I have lots of hi-res music on a NAS and I also like streaming from Spotify.

Without Spotify, I had foobar2000 on a eepc700 and I was using Bubble UPNP to stream my local music. It worked perfectly and indexing of the library when I added music was quickly done by the NAS. 

I tried sharing the music via samba , mounted in a local folder and then tried to scan it.
mopidy local scan didn't finish in 30 minutes so I gave up. I have wired gigabit ethernet, but the speed is obviously limited by the Raspberry Pi LAN.

That's why I think that andy Raspbery Pi music streamer should support both MPD and UPNP, using the Pi cpu only for decoding and playback. No indexing / serving and other stuff (that can be another toy).

UPNP via gmediarender works failry well, but switching between gmediarender and MPD (local music vs Spotify) still creates problems and sometimes the system needs to be restarted. That's annoying.

I looked at mopidy and it has MPRIS support + instructions for setting up rygel to expose it as a DLNA Media Renderer so I gave it a shot.
- both mopidy_mpris and rygel work with dbus session servers so you need to create one.
  The cheapest / fastest is to use Xvfb as described here:
  http://unix.stackexchange.com/questions/105964/launch-a-fake-minimal-x-session-for-pulseaudio-dbus

```
> Xvfb :1  -screen 0 1x1x8 &
```

Then edit /etc/rygel.conf and disable everything except MPRIS. 
MPRIS is enabled by default in rygel and I added the mopidy player config:
...
[MPRIS]

[org.mpris.MediaPlayer2.mopidy]
title=Mihai's Mopidy on @HOSTNAME@
....

The only thing you need to make sure after that is that rygel runs with mopidy credentials and it works, the new renderer appears in BubbleUPNP and it plays.

There are 2 problems I noticed:
- playback position is not displayed in BubbleUPNP
- every time a new song starts playing via rygel it plays for 1 second , restarts and then plays ok. In BubbleUPNP first time the name of the old song is displayed and then when the music restarts, the correct track is displayed.

You will notice that switching between MPD control and UPNP control works without gliches on the MPD side. Bubble UPNP controller gets confused, but it's a resumable state, nothing is crashed or blocked.

---

Crazy stuff bellow:

If you don't want to install Xvfb, you could also think of hacking mopidy_mpris and rygel to use the SystemBus instead of SessionBus.
Hacking MPRIS is easy: 
Find objects.py in mopidy_mpris and change SessionBus to SystemBus in _connect_to_dbus

```
    def _connect_to_dbus(self):
        logger.debug('Connecting to D-Bus...')
        bus_name = dbus.service.BusName(BUS_NAME, dbus.SystemBus())
        logger.info('MPRIS server connected to D-Bus')
        return bus_name
```

I found the module to be installed in /usr/local/lib/python2.7/dist-packages/mopidy_mpris/objects.py and editing + restarting mopidy had immediate effect.

rygel is another beast and I didn't had time to learn 'vala' or whatever is that language so I abandoned the SystemBus approach.
